src/plugins/mongodb/mongodb_instance_config.cpp

```diff
@@ -106,6 +108,49 @@ MongoDBInstanceConfig::MongoDBInstanceConfig(Configuration *config,
 		}
 	}
 
+	if (enabled_) {
+		std::string extra_args;
+		try {
+			extra_args = config->get_string(prefix + "args");
+		} catch (Exception &e) {}
```

Comment:
Good catch. Thanks @nlimpert for chiming in! Old habits die hard.

It doesn't actually look as nice as you write, since the _or_default variants do not come with a std::string path overload as the non-default variants do. But it's still simpler.